File: src/providers/integraicp/provider.ts
import type { Provider } from "../../core/registry.js";
import { integraIcpRoutes } from "./routes.js";
import { getCredential, listCredentials } from "./state.js";

export const integraIcpProvider: Provider = {
  name: "integraicp",

  register(app) {
    return integraIcpRoutes(app);
  },

  listItems() {
    return listCredentials().map((c) => ({
      provider: "integraicp",
      type: "signature",
      id: c.id,
      status: "AUTHENTICATED",
      createdAt: c.createdAt,
    }));
  },

  getItemDetail(id) {
    const credential = getCredential(id);
    if (!credential) return undefined;
    return { provider: "integraicp", payload: credential, webhookDeliveries: [] };
  },
};

File: internal/providers/integraicp/provider.go
package integraicp

import (
	"database/sql"
	"net/http"

	"github.com/golgimed/mimic/internal/registry"
	"github.com/golgimed/mimic/internal/shared/admin"
)

func New(db *sql.DB, faultStore *admin.Store) *registry.Provider {
	store := NewStore(db)

	return &registry.Provider{
		Name: Name,
		Register: func(mux *http.ServeMux) {
			registerRoutes(mux, store, faultStore)
		},
		ListItems: func() []registry.DashboardItem {
			credentials, err := store.ListCredentials()
			if err != nil {
				return nil
			}
			items := make([]registry.DashboardItem, 0, len(credentials))
			for _, c := range credentials {
				items = append(items, registry.DashboardItem{
					Provider:  Name,
					Type:      "signature",
					ID:        c.ID,
					Status:    "AUTHENTICATED",
					CreatedAt: c.CreatedAt,
				})
			}
			return items
		},
		GetItemDetail: func(id string) (*registry.ItemDetail, bool) {
			credential, found, err := store.GetCredential(id)
			if err != nil || !found {
				return nil, false
			}
			return &registry.ItemDetail{
				Provider:          Name,
				Payload:           credential,
				WebhookDeliveries: []registry.WebhookDeliveryView{},
			}, true
		},
	}
}
